pkg/trace/api/container_linux.go

```diff
@@ -144,9 +138,22 @@ type cgroupIDProvider struct {
 // * Datadog-Container-ID header
 // * Looks for a PID in the ctx which is used to search cgroups for a container ID.
 func (c *cgroupIDProvider) GetContainerID(ctx context.Context, h http.Header) string {
+	originInfo := origindetection.OriginInfo{ProductOrigin: origindetection.ProductOriginAPM}
+
 	// Retrieve container ID from Local Data header
-	if localData := h.Get(header.LocalData); localData != "" {
-		return c.resolveContainerIDFromLocalData(localData)
+	if localDataString := h.Get(header.LocalData); localDataString != "" {
+		var localDataErr error
+		originInfo.LocalData, localDataErr = origindetection.ParseLocalData(localDataString)
```

Comment:
nit: consider sticking to the `err` naming convention for consistency

Reply:
Sure! Done 👍